wallet/src/descriptor/mod.rs

```diff
@@ -312,7 +312,14 @@ pub(crate) fn check_wallet_descriptor(
     }
 
     if descriptor.is_multipath() {
-        return Err(DescriptorError::MultiPath);
+        // Only allow multipath descriptors with exactly 2 paths (receive and change)
+        let paths = descriptor
+            .clone()
+            .into_single_descriptors()
+            .map_err(|_| DescriptorError::MultiPath)?;
+        if paths.len() != 2 {
+            return Err(DescriptorError::MultiPath);
+        }
```

Comment:
I don't know what others think, but IMO `check_wallet_descriptor` should still reject a multipath descriptor, because we expect to be able to insert the checked descriptor directly into the Indexer, which doesn't do any extra validation.

```suggestion
         return Err(DescriptorError::Miniscript(
            miniscript::Error::BadDescriptor(
                "`check_wallet_descriptor` must not contain multipath keys".to_string(),
            ),
        ));
```

Reply:
If the point here is that `check_wallet_descriptor` should only be used to verify that the descriptors are valid for use in the indexer then should @schjonhaug put this multipath verification directly in the `create_multipath` function? then call `check_wallet_descriptor` there too on the split out non-multipath version of the descriptors?